community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ShortestPathLongerAcceptanceTest.scala

```diff
@@ -39,6 +39,43 @@ class ShortestPathLongerAcceptanceTest extends ExecutionEngineFunSuite with NewP
 
   val VERBOSE = false // Lots of debug prints
 
+  test("shortestPath with same start and end node should return zero length path with no fallback") {
+    val start = System.currentTimeMillis
+    val results = executeUsingCostPlannerOnly(
+      s"""PROFILE MATCH p = shortestPath((src:$topLeft)-[*0..]-(dst:$topLeft))
+         |RETURN nodes(p) AS nodes""".stripMargin)
+
+    dprintln(s"Query took ${(System.currentTimeMillis - start)/1000.0}s")
+    dprintln(results.executionPlanDescription())
+
+    val result = results.columnAs[List[Node]]("nodes").toList
+    debugResults(result(0).asJava.asScala.toList)
+
+    // Then
+    result.length should equal(1)
+    result(0).toSet should equal(Set(node(0, 0)))
+    results shouldNot executeShortestPathFallbackWith(minRows = 1)
+  }
+
+  test("shortestPath with same start and end node as well as predicates should resort to fallback") {
+    val start = System.currentTimeMillis
+    val results = executeUsingCostPlannerOnly(
+      s"""PROFILE MATCH p = shortestPath((src:$topLeft)-[*0..]-(dst:$topLeft))
```

Comment:
shortestPath does not allow longer minimum than 1, so it generates an immediate error `shortestPath(...) does not support a minimal length different from 0 or 1 (line 1, column 32 (offset: 31))`

We could consider (as separate work) planning shortestPath requests directly with the fallback approach for cases like that.